sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -21,7 +21,7 @@ protected DefaultAzureCredentialFactory(CredentialPipeline pipeline)
 
         public virtual TokenCredential CreateEnvironmentCredential()
         {
-            return new EnvironmentCredential(Pipeline);
+            return new EnvironmentCredential(Pipeline, new EnvironmentCredentialOptions { SendCertificateChain = true });
```

Comment:
Would this be a potentially breaking change to default to true?

Reply:
Behavior change, yes. But according to [this](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#code), not technically a breaking change because it only impacts performance.

When this is `true`, MSAL will send the entire certificate chain in the [x5c](https://datatracker.ietf.org/doc/html/rfc7517#section-4.7) header of the jwt token when authenticating with AAD. This certificate chain is necessary for AAD to perform the validate the certificate issuer. When this is `false`, I believe MSAL only sends the [x5t](https://datatracker.ietf.org/doc/html/rfc7517#section-4.8) header which contains only the certificate thumbprint which is usually much smaller.

Reply:
Actually, you're right. This is breaking. I just did a local test and discovered that if this value is `true`, it surprisingly breaks authentication for applications that still rely on thumbprint validation. Possibly an artificial limitation in MSAL or AAD.

How would you feel if I added a `DefaultAzureCredentialOptions.EnvironmentCredentialSendCertificateChain` (_name?_) property and plumbed that through to `EnvironmentCredentialOptions.SendCertificateChain`?

Or maybe remove `EnvironmentCredentialOptions.SendCertificateChain` altogether and replace the configuration functionality with an `AZURE_CLIENT_CERTIFICATE_X5C` environment variable? That would actually be much more consistent with the spirit of `EnvironmentCredential`.

Reply:
I agree that an environment variable would be the most consistent. As far as the name, I'm on the fence between something as implementation specific as `AZURE_CLIENT_CERTIFICATE_SEND_X5C` or if it should be something feature specific like `AZURE_CLIENT_CERTIFICATE_ENABLE_SNI`

Reply:
Iteration 2 takes this new environment variable approach. It feels much more natural to bundle this new setting alongside all the existing environment variables, as opposed to my initial approach that split the settings between a new `EnvironmentCredentialOptions` class and environment variables.

This new approach has a few more benefits:
- There is already a bunch of test scaffolding in this area, so it is very easy to test my change
- No change to default behavior. No breaking change.
- No need to add/set any properties in `DefaultAzureCredentialsOptions`.
  - In the case of my own application, I can simply set the environment variable (in the same place where I set all the other environment variables) in my test environment, and simple parameterless `new DefaultAzureCredential()` just works with no fuss.

Reply:
I prefer your recommended name: `AZURE_CLIENT_CERTIFIXCATE_ENABLE_SNI`. I'll wait a day or so to hear feedback from @schaabs if he has any before I make that change.